src/shims/time.rs

```diff
@@ -107,6 +112,91 @@ pub trait EvalContextExt<'mir, 'tcx: 'mir>: crate::MiriInterpCxExt<'mir, 'tcx> {
         Ok(0)
     }
 
+    // The localtime() function shall convert the time in seconds since the Epoch pointed to by
+    // timer into a broken-down time, expressed as a local time.
+    // https://linux.die.net/man/3/localtime_r
+    fn localtime_r(
+        &mut self,
+        timep: &OpTy<'tcx, Provenance>,
+        result_op: &OpTy<'tcx, Provenance>,
+    ) -> InterpResult<'tcx, Pointer<Option<Provenance>>> {
+        let this = self.eval_context_mut();
+
+        this.assert_target_os_is_unix("localtime_r");
+        this.check_no_isolation("`localtime_r`")?;
+
+        let timep = this.deref_pointer(timep)?;
+        let result = this.deref_pointer_as(result_op, this.libc_ty_layout("tm"))?;
+        let result_ptr = this.read_pointer(result_op)?;
+
+        let sec_since_epoch: i64 = this
+            .read_scalar(&timep)?
+            .to_int(this.libc_ty_layout("time_t").size)?
+            .try_into()
+            .unwrap();
+        let dt_utc: DateTime<Utc> =
+            DateTime::from_timestamp(sec_since_epoch, 0).expect("Invalid timestamp");
+        let dt: DateTime<Local> = DateTime::from(dt_utc);
+
+        let tm_sec = dt.second();
+        let tm_min = dt.minute();
+        let tm_hour = dt.hour();
+        let tm_mday = dt.day();
+        let tm_mon = dt.month0();
+        let tm_year = dt.year().saturating_sub(1900);
```

Comment:
Is it even possible for this to overflow? Why use saturating arithmetic?

Reply:
Since ``i32`` maximum value is ``2147483647``, should be impossible to overflow, at least not soon :). But I use this because the compiler will produce arithmetic side effect warning if I do 
```rust
let tm_year = dt.year() - 1900;
```
So this is mainly to eliminate the warning.

Reply:
The intended way to eliminate the warning is `checked_sub().unwrap()`